csr/csr.go

```diff
@@ -31,12 +31,12 @@ var badSignatureAlgorithms = map[x509.SignatureAlgorithm]bool{
 }
 
 var (
-	invalidPubKey       = errors.New("invalid public key in CSR")
-	unsupportedSigAlg   = errors.New("signature algorithm not supported")
-	invalidSig          = errors.New("invalid signature on CSR")
-	invalidEmailPresent = errors.New("CSR contains one or more email address fields")
-	invalidIPPresent    = errors.New("CSR contains one or more IP address fields")
-	invalidNoDNS        = errors.New("at least one DNS name is required")
+	invalidPubKey       = berrors.MalformedError("invalid public key in CSR")
```

Comment:
This function can return a variety of non-`Malformed` errors, i.e. via the calls to `pa.WillingToIssue` and `pa.checkHostlist` which prevent us from just blanket wrapping everything that comes back as a `berrors.MalformedError`. We could do a type check on return unless it was a specific set of errors returned wrap it but that seems to get into the same issues and adds a weird step we always have to remember to do after calling to this specific function.

To me it seems much cleaner to just indicate the error type at the creation site and not have to worry about if it needs special wrapping further down the line.